automated install/basic-install.sh

```diff
@@ -135,644 +151,667 @@ show_ascii_berry() {
-      i=0
-      # fuser is a program to show which processes use the named files, sockets, or filesystems
-      # So while the command is true
-      while fuser /var/lib/dpkg/lock >/dev/null 2>&1 ; do
-        # Wait half a second
-        sleep 0.5
-        # and increase the iterator
-        ((i=i+1))
-      done
-      # Always return success, since we only return if there is no
-      # lock (anymore)
-      return 0
-    }
 
-# If apt-get is not found, check for rpm to see if it's a Red Hat family OS
-elif command -v rpm &> /dev/null; then
-  # Then check if dnf or yum is the package manager
-  if command -v dnf &> /dev/null; then
-    PKG_MANAGER="dnf"
-  else
-    PKG_MANAGER="yum"
-  fi
+  # the package manager itself
+  PKG_MANAGER=""
+  # the command used for updating the package cache
+  UPDATE_PKG_CACHE=""
+  # the command used for installing new packages
+  PKG_INSTALL=""
+  # ?
+  PKG_COUNT=""
+
+  # dependencies for the installer, Pi-hole and the web dashboard
+  INSTALLER_DEPS=""
+  PIHOLE_DEPS=""
+  PIHOLE_WEB_DEPS=""
+
+  # the user, group and config file for the web server
+  LIGHTTPD_USER=""
+  LIGHTTPD_GROUP=""
+  LIGHTTPD_CFG=""
+
+
+  # Debian-family (Debian, Raspbian, Ubuntu)
+  # If apt-get is installed, then we know it's part of the Debian family.
+  if command -v apt-get &> /dev/null; then
+
+    PKG_MANAGER="apt-get"
+    UPDATE_PKG_CACHE="${PKG_MANAGER} update"
+    PKG_INSTALL=("${PKG_MANAGER}" --yes --no-install-recommends install)
+
+    # grep -c will return 1 retVal on 0 matches, block this throwing the set -e with an OR TRUE
+    PKG_COUNT="${PKG_MANAGER} -s -o Debug::NoLocking=true upgrade | grep -c ^Inst || true"
+
+    # Debian 7 doesn't have iproute2. Use it if the dryrun install is successful, otherwise use iproute.
+    local iproutePkg
+    if "${PKG_MANAGER}" install --dry-run iproute2 > /dev/null 2>&1; then
+      iproutePkg="iproute2"
+    else
+      iproutePkg="iproute"
+    fi
+
+    # We prefer the php metapackage if it's there, otherwise use the php5 packages.
+    if "${PKG_MANAGER}" install --dry-run php > /dev/null 2>&1; then
+      phpVer="php"
+    else
+      phpVer="php5"
+    fi
+
+    # Determine the available version of `php-sqlite`.
+    if "${PKG_MANAGER}" install --dry-run "${phpVer}"-sqlite3 > /dev/null 2>&1; then
+      phpSqlite="sqlite3"
+    else
+      phpSqlite="sqlite"
+    fi
+
+    INSTALLER_DEPS=(apt-utils dialog debconf dhcpcd5 git "${iproutePkg}" whiptail)
+    PIHOLE_DEPS=(bc cron curl dnsutils iputils-ping lsof netcat psmisc sudo unzip wget idn2 sqlite3 libcap2-bin dns-root-data resolvconf)
+    PIHOLE_WEB_DEPS=(lighttpd "${phpVer}"-common "${phpVer}"-cgi "${phpVer}"-"${phpSqlite}")
+
+    LIGHTTPD_USER="www-data"
+    LIGHTTPD_GROUP="www-data"
+    LIGHTTPD_CFG="lighttpd.conf.debian"
+
+    # Waits until the dpkg lock is free and packages can be installed/configured.
+    test_dpkg_lock() {
+        # Keep track of iterations. TODO: Why?
+        local i=0
```

Comment:
If it's not being used then someone deleted the function call. It's important to have as it checks to see if `dpkg` has locked the database and thus prevents any `apt install` commands from succeeding. Cloud-init will call for an `apt update` on first boot and if we try to install during the period that the database is locked it will fail in spectacular ways.